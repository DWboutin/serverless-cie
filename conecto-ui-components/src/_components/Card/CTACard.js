import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import omit from 'omit.js'

const CTACard = ({
  className,
  children,
  image,
  button,
  withOverlap,
  ...props
}) => {
  const classes = classNames({
    'to-card': true,
    'to-card-cta': true,
    'with-overlap': withOverlap,
    [`${className}`]: className,
  })
  const componentProps = omit(props, [
    'className',
    'children',
    'image',
    'button',
    'withOverlap',
  ])

  return (
    <div className={classes} {...componentProps}>
      <div className="content">{children}</div>
      <div className="cta">
        <div className="image">
          <div className="image-wrap">{image}</div>
          {button}
        </div>
      </div>
    </div>
  )
}

CTACard.defaultProps = {
  withOverlap: false,
}

CTACard.propTypes = {
  withOverlap: PropTypes.bool.isRequired,
  button: PropTypes.element.isRequired,
  children: PropTypes.any.isRequired,
  image: PropTypes.node.isRequired,
}

export default CTACard
