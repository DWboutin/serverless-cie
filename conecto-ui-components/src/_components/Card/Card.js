import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import omit from 'omit.js'

const Card = ({ className, withIcon, type, icon, children, ...props }) => {
  const classes = classNames({
    'to-card': true,
    [`to-card-${type}`]: type,
    ['to-card-with-icon']: withIcon,
    [`${className}`]: className,
  })
  const componentProps = omit(props, [
    'className',
    'withIcon',
    'type',
    'icon',
    'children',
  ])

  return (
    <div className={classes} {...componentProps}>
      {withIcon && <div className="icon-wrapper">{icon}</div>}
      <div className="to-card-wrap">{children}</div>
    </div>
  )
}

Card.defaultProps = {
  withIcon: false,
  type: 'bordered',
}

Card.propTypes = {
  type: PropTypes.oneOf(['bordered', 'shadowed']).isRequired,
  withIcon: PropTypes.bool.isRequired,
  children: PropTypes.any.isRequired,
  icon: PropTypes.element,
}

export default Card
