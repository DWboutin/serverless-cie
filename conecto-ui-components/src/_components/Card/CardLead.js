import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import omit from 'omit.js'

import Icon from '../Icon'

const CardLead = ({
  className,
  logo,
  buttons,
  checkboxes,
  requester,
  img,
  latitude,
  longitude,
  opportunityId,
  opportunityDate,
  opportunityRemainingPlace,
  labelCity,
  labelJobType,
  labelJobTypeSpecific,
  labelRequester,
  labelContactPreference,
  valueCity,
  valueJobType,
  valueJobTypeSpecific,
  valueMoreInfosAccordion,
  ...props
}) => {
  const classes = classNames({
    'to-card to-card-lead': true,
    'has-accordion': valueMoreInfosAccordion,
    [`${className}`]: className,
  })
  const componentProps = omit(props, [
    'className',
    'logo',
    'buttons',
    'checkboxes',
    'requester',
    'img',
    'latitude',
    'longitude',
    'opportunityId',
    'opportunityDate',
    'labelCity',
    'labelJobType',
    'labelJobTypeSpecific',
    'labelRequester',
    'labelContactPreference',
    'valueCity',
    'valueJobType',
    'valueJobTypeSpecific',
    'valueMoreInfosAccordion',
  ])

  return (
    <div className={classes} {...componentProps}>
      <header className="card-lead-header">
        <div className="logo">{logo}</div>
        <div className="header-info">
          <div className="opportunity-id">{opportunityId}</div>
          <div className="opportunity-time">{opportunityDate}</div>
        </div>
      </header>
      <div className="to-card-wrap">
        <div className="to-card-map">
          <a
            href={`https://www.google.com/maps/search/?api=1&query=${latitude},${longitude}`}
            target="_blank"
          >
            {img}
          </a>
        </div>
        <div className="to-card-content">
          <div className="opportunity-city">
            <div className="label">{labelCity}</div>
            <div className="value">
              <Icon type="marker" />
              <span>{valueCity}</span>
            </div>
          </div>
          <div className="job-type">
            <div className="label">{labelJobType}</div>
            <div className="value">
              <span>{valueJobType}</span>
            </div>
          </div>
          <div className="job-type-specific">
            <div className="label">{labelJobTypeSpecific}</div>
            <div className="value">
              <span>{valueJobTypeSpecific}</span>
            </div>
          </div>
        </div>
        {valueMoreInfosAccordion}
      </div>
      <footer className="card-lead-footer">
        <div className="contact-infos">
          <div className="info">
            <div className="label">{labelRequester}</div>
            <div className="value">
              <span>{requester}</span>
            </div>
          </div>
          <div className="info">
            <div className="label">{labelContactPreference}</div>
            <div className="value">{checkboxes}</div>
          </div>
        </div>
        <div className="contact-actions">{buttons}</div>
      </footer>
    </div>
  )
}

CardLead.defaultProps = {
  valueMoreInfosAccordion: null,
}

CardLead.propTypes = {
  logo: PropTypes.node.isRequired,
  buttons: PropTypes.node.isRequired,
  checkboxes: PropTypes.node.isRequired,
  img: PropTypes.node.isRequired,
  requester: PropTypes.string.isRequired,
  opportunityId: PropTypes.string.isRequired,
  opportunityDate: PropTypes.string.isRequired,
  labelCity: PropTypes.string.isRequired,
  labelJobType: PropTypes.string.isRequired,
  labelJobTypeSpecific: PropTypes.string.isRequired,
  labelRequester: PropTypes.string.isRequired,
  labelContactPreference: PropTypes.string.isRequired,
  valueCity: PropTypes.string.isRequired,
  valueJobType: PropTypes.string.isRequired,
  valueJobTypeSpecific: PropTypes.string.isRequired,
  valueMoreInfosAccordion: PropTypes.any,
  latitude: PropTypes.any.isRequired,
  longitude: PropTypes.any.isRequired,
}

export default CardLead
