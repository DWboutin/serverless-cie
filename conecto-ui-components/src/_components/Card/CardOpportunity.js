import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import omit from 'omit.js'

import Icon from '../Icon'

const CardOpportunity = ({
  className,
  img,
  logo,
  buttons,
  accepted,
  opportunityId,
  opportunityDate,
  opportunityRemainingPlace,
  labelCity,
  labelJobType,
  labelJobTypeSpecific,
  valueCity,
  valueJobType,
  valueJobTypeSpecific,
  valueMoreInfosAccordion,
  ...props
}) => {
  const classes = classNames({
    'to-card to-card-opportunity': true,
    'has-accordion': valueMoreInfosAccordion,
    [`${className}`]: className,
  })
  const componentProps = omit(props, [
    'className',
    'img',
    'logo',
    'buttons',
    'accepted',
    'opportunityId',
    'opportunityDate',
    'opportunityRemainingPlace',
    'labelCity',
    'labelJobType',
    'labelJobTypeSpecific',
    'valueCity',
    'valueJobType',
    'valueJobTypeSpecific',
    'valueMoreInfosAccordion',
    'onAcceptation',
    'onReject',
  ])

  return (
    <div className={classes} {...componentProps}>
      <header className="card-opporunity-header">
        <div className="logo">{logo}</div>
        <div className="header-info">
          <div className="opportunity-id">{opportunityId}</div>
          <div className="opportunity-time">{opportunityDate}</div>
        </div>
        {opportunityRemainingPlace && !accepted && (
          <span className="remaining-places">{opportunityRemainingPlace}</span>
        )}
      </header>
      <div className="to-card-wrap">
        {img && <div className="to-card-map">{img}</div>}
        <div className="to-card-content">
          <div className="opportunity-city">
            <div className="label">{labelCity}</div>
            <div className="value">
              <Icon type="marker" />
              <span>{valueCity}</span>
            </div>
          </div>
          <div className="opportunity-job">
            <div className="job-type">
              <div className="label">{labelJobType}</div>
              <div className="value">
                <span>{valueJobType}</span>
              </div>
            </div>
            <div className="job-type-specific">
              <div className="label">{labelJobTypeSpecific}</div>
              <div className="value">
                <span>{valueJobTypeSpecific}</span>
              </div>
            </div>
          </div>
        </div>
        {valueMoreInfosAccordion}
      </div>
      <footer className="card-opporunity-footer">{buttons}</footer>
    </div>
  )
}

CardOpportunity.defaultProps = {
  valueMoreInfosAccordion: null,
}

CardOpportunity.propTypes = {
  img: PropTypes.node.isRequired,
  logo: PropTypes.node.isRequired,
  accepted: PropTypes.bool.isRequired,
  buttons: PropTypes.node.isRequired,
  opportunityId: PropTypes.string.isRequired,
  opportunityDate: PropTypes.string.isRequired,
  opportunityRemainingPlace: PropTypes.string.isRequired,
  labelCity: PropTypes.string.isRequired,
  labelJobType: PropTypes.string.isRequired,
  labelJobTypeSpecific: PropTypes.string.isRequired,
  valueCity: PropTypes.string.isRequired,
  valueJobType: PropTypes.string.isRequired,
  valueJobTypeSpecific: PropTypes.string.isRequired,
  valueMoreInfosAccordion: PropTypes.any,
}

export default CardOpportunity
