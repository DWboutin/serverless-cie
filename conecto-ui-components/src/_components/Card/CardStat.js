import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

const CardStat = ({ className, type, icon, title, children }) => {
  const classes = classNames({
    'to-card': true,
    [`to-card-${type}`]: type,
    'to-card-stat': true,
    [`${className}`]: className,
  })

  return (
    <div className={classes}>
      <div className="header">
        <span className="circle">{icon}</span>
        <span className="title">{title}</span>
      </div>
      <div className="to-card-wrap">{children}</div>
    </div>
  )
}

CardStat.defaultProps = {
  type: 'bordered',
}

CardStat.propTypes = {
  type: PropTypes.oneOf(['bordered', 'shadowed']).isRequired,
  children: PropTypes.any.isRequired,
  icon: PropTypes.element.isRequired,
}

export default CardStat
