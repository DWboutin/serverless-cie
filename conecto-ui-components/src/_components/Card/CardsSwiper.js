import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import Swiper from 'swiper/dist/js/swiper.js'
import 'swiper/dist/css/swiper.min.css'

class CardsSwiper extends React.Component {
  constructor(props) {
    super(props)

    this.classesPrefix = 'to-cards-swiper'

    this.swiper = null
    this.defaultOptions = {
      speed: 500,
      slidesPerView: 3,
      spaceBetween: 40,
      // simulateTouch: false,
      breakpoints: {
        576: {
          slidesPerView: 'auto',
          spaceBetween: 20,
        },
        768: {
          slidesPerView: 'auto',
          spaceBetween: 20,
        },
        992: {
          slidesPerView: 3,
          spaceBetween: 20,
        },
      },
    }
  }

  componentDidMount() {
    const { id } = this.props

    this.swiper = new Swiper(`#${id}`, {
      ...this.defaultOptions,
      ...this.props.options,
    })
  }

  componentWillUnmount() {
    this.swiper.destroy(true, true)
  }

  componentDidUpdate() {
    this.swiper.update()
  }

  render() {
    const { id, className, withIcons } = this.props
    const classes = classNames({
      [this.classesPrefix]: true,
      'swiper-container': true,
      'with-icons': withIcons,
      [`${className}`]: className,
    })

    return (
      <div id={id} className={classes}>
        <div className="swiper-wrapper">
          {React.Children.map(this.props.children, (child, index) => (
            <div className="swiper-slide">{child}</div>
          ))}
        </div>
      </div>
    )
  }
}

CardsSwiper.defaultProps = {
  withIcons: false,
  options: {},
}

CardsSwiper.propTypes = {
  id: PropTypes.string.isRequired,
  withIcons: PropTypes.bool.isRequired,
  options: PropTypes.object.isRequired,
}

export default CardsSwiper
