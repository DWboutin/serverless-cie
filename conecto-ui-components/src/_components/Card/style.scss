@import '../_base/variables';

.to-card {
  display: block;
  position: relative;
  border-radius: 10px;
  padding: 40px;

  &.to-card-bordered {
    border: 1px solid $veryPaleBrown;
  }

  &.to-card-shadowed {
    box-shadow: 0px 0px 15px rgba(85, 62, 65, 0.15);
  }

  .icon-wrapper {
    display: inline-block;
    position: absolute;
    top: -31px;
    left: 40px;
    width: 62px;
    height: 62px;
    border-radius: 50%;
    background-color: $paleGrey;
    padding: 13px 0 0 17px;
  }

  &:not(.to-card-with-icon) {
    overflow: hidden;
  }

  &.to-card-with-icon {
    .icon-wrapper .to-icon {
      display: block;
      position: relative;
      width: 34px;
      height: 34px;

      svg {
        width: 100%;
        height: 100%;
      }
    }
  }

  &.to-card-cta {
    display: flex;
    flex-flow: wrap;
    overflow: visible;
    padding: 0;
    border: 0;

    &.with-overlap {
      .cta {
        .image {
          .image-wrap {
            height: 120%;
          }
        }
      }
    }

    &>div {
      flex: 0 0 50%;
    }

    .content {
      padding: 38px 32px 36px;
      border-top: 1px solid $veryPaleBrown;
      border-left: 1px solid $veryPaleBrown;
      border-bottom: 1px solid $veryPaleBrown;
      border-radius: 10px 0 0 10px;
    }

    .cta {
      position: relative;
      background-color: $paleGreen;
      border-top: 1px solid $paleGreen;
      border-right: 1px solid $paleGreen;
      border-bottom: 1px solid $paleGreen;
      border-radius: 0 10px 10px 0;
      align-items: center;
      justify-content: center;

      .image {
        display: flex;
        position: relative;
        align-items: center;
        justify-content: center;
        height: 100%;
        text-align: center;

        .image-wrap {
          display: block;
          position: absolute;
          bottom: 0px;
          height: 100%;
          width: 100%;
          overflow: hidden;
          border-radius: 0 10px 10px 0;

          svg {
            display: block;
            position: absolute;
            min-width: 100%;
            max-height: 100%;
            margin-left: auto;
            margin-right: auto;
            left: 0;
            right: 0;
            bottom: 0px;
          }
        }
      }
    }
  }

  &.to-card-opportunity {
    padding: 0;
    border: 1px solid $veryPaleBrown;

    header {
      display: flex;
      position: relative;
      flex: 0 1 auto;
      justify-content: center;
      align-items: center;
      padding: 30px;
      border-bottom: 1px solid $veryPaleBrown;

      &>div {
        display: inline-block;
        flex: 0 0 50%;

        &:last-child {
          text-align: right;
        }
      }

      .to-logo-wrapper {
        width: 100%;
      }

      .opportunity-id {
        font-family: StyreneA, Arial, Helvetica, sans-serif;
        font-size: sizeToRem(14);
        font-weight: normal;
        line-height: normal;
        letter-spacing: 0.02em;
        margin-bottom: 4px;
        text-align: right;
      }
      .opportunity-time {
        font-family: Sailec, Arial, Helvetica, sans-serif;
        font-size: sizeToRem(12);
        line-height: sizeToRem(18);
        color: $paleBrown;
        text-align: right;
      }
      .remaining-places {
        display: inline-block;
        position: absolute;
        bottom: 0px;
        left: 50%;
        transform: translate(-50%, 50%);
        font-family: Sailec, Arial, Helvetica, sans-serif;
        font-size: sizeToRem(12);
        line-height: sizeToRem(18);
        color: $paleBrown;
        border-radius: 5px;
        text-align: center;
        padding: 10px 15px 10px;
        border: 1px solid $veryPaleBrown;
        background-color: #fff;
        z-index: 1;
      }
    }

    footer {
      padding: 30px;
      text-align: right;
      border-top: 1px solid $veryPaleBrown;

      .to-btn {
        margin-left: 20px;
      }
    }

    &.has-accordion {
      .to-card-wrap {
        .to-card-content {
          padding: 30px 30px 10px;
        }
      }
    }

    .to-card-wrap {
      .to-card-content {
        padding: 30px;
      }

      .to-card-map {
        display: block;
        position: relative;
        text-align: center;
        height: 180px;
        min-width: 100%;
        overflow: hidden;
        border-bottom: 1px solid $veryPaleBrown;

        img {
          position: absolute;
          top: -9999px;
          bottom: -9999px;
          left: -9999px;
          right: -9999px;
          margin: auto;
          min-width: 100%;
        }
      }

      .to-accordion {
        .to-accordion-panel {
          .to-accordion-panel-title {
            padding-left: 30px;
            padding-right: 30px;

            .arrow {
              right: 30px;
            }
          }
          .to-accordion-panel-content {
            .content-wrap {
              padding-left: 30px;
              padding-right: 30px;
            }
          }
        }
      }

      .label {
        font-family: Sailec, Arial, Helvetica, sans-serif;
        font-size: sizeToRem(12);
        line-height: sizeToRem(18);
        color: $paleBrown;
      }

      .value {
        display: block;
        position: relative;
        font-family: Sailec, Arial, Helvetica, sans-serif;
        font-size: sizeToRem(14);
        line-height: sizeToRem(21);
        color: $darkBrown;
        
        .to-icon {
          display: inline-block;
          position: relative;
          margin-right: 5px;
          top: -2px;
        }
      }

      .opportunity-city,
      .opportunity-job {
        padding-top: 10px;
        padding-bottom: 10px;
      }

      .opportunity-job {
        display: flex;
        position: relative;
        flex: 0 1 auto;

        &>div {
          display: inline-block;
          flex: 0 0 50%;
        }
      }
    }
  }

  &.to-card-lead {
    padding: 0;
    border: 1px solid $veryPaleBrown;

    header {
      display: flex;
      position: relative;
      flex: 0 1 auto;
      justify-content: center;
      align-items: center;
      padding: 30px;
      border-bottom: 1px solid $veryPaleBrown;

      &>div {
        display: inline-block;
        flex: 0 0 50%;

        &:last-child {
          text-align: right;
        }
      }

      .to-logo-wrapper {
        width: 100%;
      }

      .opportunity-id {
        font-family: StyreneA, Arial, Helvetica, sans-serif;
        font-size: sizeToRem(18);
        font-weight: normal;
        line-height: normal;
        letter-spacing: 0.02em;
        margin-bottom: 4px;
        text-align: right;
      }
      .opportunity-time {
        font-family: Sailec, Arial, Helvetica, sans-serif;
        font-size: sizeToRem(12);
        line-height: sizeToRem(18);
        color: $paleBrown;
        text-align: right;
      }
    }

    footer {
      display: flex;
      position: relative;
      flex: 0 1 auto;
      border-top: 1px solid $veryPaleBrown;


      &>div {
        display: inline-block;
        flex: 0 0 50%;
        padding: 30px;

        &:first-child {
          border-right: 1px solid $veryPaleBrown;
        }

        .info {
          margin-bottom: 20px;

          .to-checkbox {
            margin-right: 10px;

            &:last-child {
              margin-right: 0px;
            }
          }

          &:last-child {
            margin-bottom: 0px;
          }
        }
      }
    }

    .contact-actions {
      text-align: center;

      &>div {
        margin-bottom: 10px;

        &:last-child {
          margin-bottom: 0px;
        }
      }
    }

    .label {
      font-family: Sailec, Arial, Helvetica, sans-serif;
      font-size: sizeToRem(12);
      line-height: sizeToRem(18);
      color: $paleBrown;
      margin-bottom: 5px;
    }

    .value {
      display: block;
      position: relative;
      font-family: Sailec, Arial, Helvetica, sans-serif;
      font-size: sizeToRem(14);
      line-height: sizeToRem(21);
      color: $darkBrown;

      .to-icon {
        display: inline-block;
        position: relative;
        margin-right: 5px;
        top: -2px;
      }
    }

    &.has-accordion {
      .to-card-wrap {
        .to-card-content {
          padding: 30px 30px 10px;
        }
      }
    }

    .to-card-wrap {
      .to-card-map {
        display: block;
        position: relative;
        text-align: center;
        height: 250px;
        min-width: 100%;
        overflow: hidden;

        img {
          position: absolute;
          top: -9999px;
          bottom: -9999px;
          left: -9999px;
          right: -9999px;
          margin: auto;
          min-width: 100%;
        }
      }

      .to-accordion {
        .to-accordion-panel {
          .to-accordion-panel-title {
            padding-left: 30px;
            padding-right: 30px;

            .arrow {
              right: 30px;
            }
          }
          .to-accordion-panel-content {
            .content-wrap {
              padding-left: 30px;
              padding-right: 30px;
            }
          }
        }
      }

      .to-card-content {
        display: flex;
        position: relative;
        flex: 0 1 auto;
        padding: 30px;

        & > div {
          display: inline-block;
          flex: 0 0 33%;
        }
      }

      .to-accordion {
        .to-accordion-panel {
          .to-accordion-panel-title {
            padding-left: 30px;
            padding-right: 30px;

            .arrow {
              right: 30px;
            }
          }
          .to-accordion-panel-content {
            .content-wrap {
              padding-left: 30px;
              padding-right: 30px;
            }
          }
        }
      }

      .opportunity-job {
        display: flex;
        position: relative;
        flex: 0 1 auto;

        &>div {
          display: inline-block;
          flex: 0 0 50%;
        }
      }
    }
  }

  &.to-card-stat {
    text-align: center;
    padding: 12px 14px 18px;

    .header {
      text-align: left;

      .circle {
        display: inline-block;
        position: relative;
        width: 40px;
        height: 40px;
        background-color: $paleGrey;
        border-radius: 50%;
        text-align: center;

        i {
          font-size: 20px;
          line-height: 40px;
          color: $green;
        }
      }

      .title {
        margin-left: 12px;
        font-family: Sailec, Arial, Helvetica, sans-serif;
        font-size: 14px;
        line-height: 21px;
      }
    }
  }
}

.to-cards-swiper {
  overflow: hidden;

  .swiper-wrapper {
    .swiper-slide:last-child {
      margin-right: 0px !important;
    }
  }

  &.with-icons {
    padding-top: 30px;
  }
}

@media (min-width: $screen-md) and (max-width: $screen-lg - 1) {
  .to-cards-swiper {
    .swiper-wrapper {
      .swiper-slide {
        width: 45%;
      }
    }
  }
}

@media (min-width: $screen-sm) and (max-width: $screen-md - 1) {
  .to-cards-swiper {
    .swiper-wrapper {
      .swiper-slide {
        width: 90%;
      }
    }
  }

  .to-card {
    &.to-card-cta {
      flex-direction: column;

      &>div {
        flex: 1 0 50%;
      }
      .content {
        padding: 38px 32px 36px;
        border-top: 1px solid $veryPaleBrown;
        border-left: 1px solid $veryPaleBrown;
        border-right: 1px solid $veryPaleBrown;
        border-bottom: 0px solid $veryPaleBrown;
        border-radius: 10px 10px 0 0;
      }

      .cta {
        position: relative;
        background-color: $paleGreen;
        border-top: 0px solid $paleGreen;
        border-right: 1px solid $paleGreen;
        border-left: 1px solid $paleGreen;
        border-bottom: 1px solid $paleGreen;
        border-radius: 0 0 10px 10px;

        .image {
          padding: 38px 32px 36px;
        }
      }
    }
  }
}

@media (max-width: $screen-sm) {
  .to-cards-swiper {
    .swiper-wrapper {
      .swiper-slide {
        width: 90%;
      }
    }
  }

  .to-card {
    &.to-card-cta {
      flex-direction: column;

      &>div {
        flex: 1 0 50%;
      }
      .content {
        padding: 38px 32px 36px;
        border-top: 1px solid $veryPaleBrown;
        border-left: 1px solid $veryPaleBrown;
        border-right: 1px solid $veryPaleBrown;
        border-bottom: 0px solid $veryPaleBrown;
        border-radius: 10px 10px 0 0;
      }

      .cta {
        position: relative;
        background-color: $paleGreen;
        border-top: 0px solid $paleGreen;
        border-right: 1px solid $paleGreen;
        border-left: 1px solid $paleGreen;
        border-bottom: 1px solid $paleGreen;
        border-radius: 0 0 10px 10px;

        .image {
          padding: 38px 32px 36px;
        }

        .image-wrap {
          max-height: 110%;

          svg {
            height: 100%;
          }
        }
      }
    }
  }
}