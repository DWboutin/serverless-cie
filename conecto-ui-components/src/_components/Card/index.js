import Card from './Card'
import CardsSwiper from './CardsSwiper'
import CTACard from './CTACard'
import CardOpportunity from './CardOpportunity'
import CardLead from './CardLead'
import CardStat from './CardStat'

export {
  Card as default,
  CardsSwiper,
  CTACard,
  CardOpportunity,
  CardLead,
  CardStat,
}
