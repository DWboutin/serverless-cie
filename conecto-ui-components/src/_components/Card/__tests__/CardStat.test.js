import React from 'react'
import { shallow, mount } from 'enzyme'

import { CardStat } from '../../Card'
import Icon from '../../Icon'

describe('CardStat', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(
      <CardStat icon={<Icon />} title="Test title card">
        Test 1
      </CardStat>
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('should match snapshot with shadow', () => {
    const wrapper = shallow(
      <CardStat icon={<Icon />} title="Test title card" type="shadowed">
        Test 1
      </CardStat>
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('should have the prefix class', () => {
    const wrapper = shallow(
      <CardStat icon={<Icon />} title="Test title card">
        Test 1
      </CardStat>
    )

    expect(wrapper.hasClass('to-card-stat')).toEqual(true)
  })
})
