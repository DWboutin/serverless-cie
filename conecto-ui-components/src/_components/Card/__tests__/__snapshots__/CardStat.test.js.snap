// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardStat should match snapshot 1`] = `
<div
  className="to-card to-card-bordered to-card-stat"
>
  <div
    className="header"
  >
    <span
      className="circle"
    >
      <Icon
        type="conecto"
      />
    </span>
    <span
      className="title"
    >
      Test title card
    </span>
  </div>
  <div
    className="to-card-wrap"
  >
    Test 1
  </div>
</div>
`;

exports[`CardStat should match snapshot with shadow 1`] = `
<div
  className="to-card to-card-shadowed to-card-stat"
>
  <div
    className="header"
  >
    <span
      className="circle"
    >
      <Icon
        type="conecto"
      />
    </span>
    <span
      className="title"
    >
      Test title card
    </span>
  </div>
  <div
    className="to-card-wrap"
  >
    Test 1
  </div>
</div>
`;
